### ISSUE TITLE
require doesn't pickup on newly created json files

### ISSUE THREAD

**Issue by jklarenbeek** [parcel-bundler/parcel#2208]
# 🐛 bug report

To be sure I used the parceljs getting started guide and added a data subfolder with two files in it:

./data/index.js
```js
const files = require('./*.json');
const modules = {};

for (const fileName in files) {
  if (fileName === 'index') continue;
  if (fileName.startsWith('_')) continue;
  if (fileName.startsWith('.')) continue;

  modules[fileName] = files[fileName];
}

export default modules;
```

and 
./data/test1.json
```js
{
    "type": "integer",
    "enum": [ 1, 2, 3, 4 ]
}
```
then I changed the root ./index.js file as follows
```js
import jsondata from './data';

console.log('hello world');
console.log(jsondata);
```
Then when all files are saved, start parcel with parcel index.html and the console will show hello world with a object on the next line containing test1. That's correct!

Now keep parcel running and create a new json file ./data/test2.json with some arbitrary information like
```js
{
    "type": "object",
    "properties": {
        "type": "string"
    }
}
```

But nothing happens. This could be happening for some understandable reason, but when editing the ./data/index.js file it should be picked up.

However, this is not the case. :(

## 🎛 Configuration (.babelrc, package.json, cli command)

```js
{
  "name": "parcel-test",
  "version": "1.0.0",
  "description": "test if json files are automagically added to the build config",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}
```

## 🤔 Expected Behavior

The object returned from the require command should return the newly added json file.

## 😯 Current Behavior

parcel/require doesn't pick up on the newly created json file.

## 💁 Possible Solution

stop and start parcel again.

## 🔦 Context

I want to quickly add or remove json files in a subfolder. I think parcel is only hot reloading js files but forgets to do the same for json files. Besides, it seems that require is cached and not updated on those json files.

## 💻 Code Sample

Please see my bug report above with code snippets.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.10.3
| Node             | 10.12.0
| npm/Yarn         | 6.4.1
| Operating System | win10x64

**Comment by jklarenbeek** [comment]
So I jumped into the parcel code again and found that the JSONAsset.js is testing the file extension for json5 instead of json. No worries, I did the same as above but then with the json5 to no avail.

Also I'm a little bit confused, I found in Parser.js the the ./asset/JSONAsset is registered with the 2 extensions: json, json5. But the parse method in in JSONAsset explicitly checks for the json5 extension?!

Anyway, parcel is a little to big to figure out within my time limit, where the hmr is checking for file changes. (at the moment)
